modules/api/.golangci.yml

```diff
@@ -53,67 +51,91 @@ linters:
     - predeclared
     - promlinter
     - staticcheck
-    - tenv
     - unconvert
     - unused
     - wastedassign
     - whitespace
-  disable-all: true
-
-# NOTE: Do not use commas in the exclude patterns, or else the regex will be
-# split and you will be sad: https://github.com/golangci/golangci-lint/issues/665
-issues:
-  exclude:
-    # gocyclo
-    - cyclomatic complexity [0-9]+ of func `initTestEndpoint` is high
-    - cyclomatic complexity [0-9]+ of func `GetAPIV2NodeCloudSpec` is high
-    - loop variable tc captured by func literal
-    # TODO (sig-api): fix the following issues
-    - cfg.EndpointResolverWithOptions is deprecated
-  exclude-files:
-    # This package was forked from upstream (in #9826)
-    # and we want to keep it as close to upstream as possible.
-    - modules/api/pkg/provider/cloud/eks/authenticator
-
-linters-settings:
-  depguard:
+  settings:
+    depguard:
+      rules:
+        main:
+          deny:
+            - pkg: io/ioutil
+              desc: https://go.dev/doc/go1.16#ioutil
+    importas:
+      alias:
+        # KKP
+        - pkg: k8c.io/kubermatic/sdk/v2/api/v1
+          alias: apiv1
+        - pkg: k8c.io/kubermatic/sdk/v2/api/v2
+          alias: apiv2
+        - pkg: k8c.io/kubermatic/sdk/v2/apis/apps.kubermatic/v1
+          alias: appskubermaticv1
+        - pkg: k8c.io/kubermatic/sdk/v2/apis/kubermatic/v1
+          alias: kubermaticv1
+        - pkg: k8c.io/kubermatic/v2/pkg/util/errors
+          alias: utilerrors
+        - pkg: k8c.io/dashboard/v2/pkg/provider/auth/types
+          alias: authtypes
+        # Kubernetes
+        - pkg: k8s.io/api/(\w+)/(v[\w\d]+)
+          alias: $1$2
+        - pkg: k8s.io/apimachinery/pkg/apis/meta/v1
+          alias: metav1
+        - pkg: k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1
+          alias: apiextensionsv1
+        - pkg: k8s.io/apimachinery/pkg/api/errors
+          alias: apierrors
+        - pkg: k8s.io/apimachinery/pkg/util/errors
+          alias: kerrors
+        # Controller Runtime
+        - pkg: sigs.k8s.io/controller-runtime/pkg/client
+          alias: ctrlruntimeclient
+        # Misc
+        - pkg: k8c.io/machine-controller/sdk/apis/cluster/v1alpha1
+          alias: clusterv1alpha1
+        - pkg: github.com/Masterminds/semver/v3
+          alias: semverlib
+      no-unaliased: true
+    staticcheck:
+      checks:
+        - all
+        # disable "error strings should not be capitalized":
+        # but often errors start with k8s kinds and it's annoying to rephrase every error
+        - '-ST1005'
+        # very opinionated naming conventions
+        - '-ST1003'
```

Comment:
tiny nitpick from my side: Can we expand the comment, as to what conventions we don't like?
Asking because this check includes for different things https://staticcheck.dev/docs/checks#ST1003

Reply:
I misunderstood, you cannot choose between the 4 different ones. It is a all or nothing rule.